### DISCUSSION TITLE
SpiralLinux 12.231001 release

### DISCUSSION THREAD

**Discussion by geckolinux** [SpiralLinux/SpiralLinux-project#307] [Announcements]
![SpiralLinux-transparent-small](https://user-images.githubusercontent.com/2692138/172185018-77411974-5e2b-43b6-aed8-42f492047cd6.png)

### Download from: https://spirallinux.github.io
> ⚠️ ***Important note:*** Rufus, LinuxLive USB Creator (LiLi), Ventoy, and Unetbootin are **NOT** recommended by SpiralLinux for creating the bootable USB device due to many reports of failure.

The SpiralLinux project is pleased to announce its 12.231001 release built directly from Debian 12, with the addition of a newer kernel from Debian Backports for newer hardware compatibility. SpiralLinux offers live installable ISO images for the Cinnamon (5.6.8), XFCE (4.18), Gnome (43.6), Plasma (5.27), Mate (1.26), Budgie (10.7.1), and LXQt (1.2.0) desktop environments.

## About SpiralLinux

SpiralLinux is a selection of Linux spins built from Debian GNU/Linux, with a focus on simplicity and out-of-the-box usability across all the major desktop environments. SpiralLinux serves as an alternative live installation method for a highly reliable customized Debian system using only official Debian package repositories. Please visit the official [SpiralLinux website](https://spirallinux.github.io/) to download it and learn more about its unique features and reasons for existence.

## Notable changes in SpiralLinux 12.231001
- **Low-latency Pipewire configuration:** For this release all editions of SpiralLinux are now using Pipewire with a proper out-of-the-box configuration for low-latency (5.3ms by default) JACK-compatible live audio applications even on standard consumer hardware. The default configuration is unique to SpiralLinux and includes overrides to the default Pipewire configuration as well as a number of low-level system tweaks as recommended by [rtcqs](https://codeberg.org/rtcqs/rtcqs). This makes SpiralLinux an excellent base system for professional live audio production JACK workflows while simultaneously offering compatibility with normal PulseAudio consumer applications. To take full advantage of the low-latency JACK compatibility simply run `sudo systemctl enable cpupower` to enable the optimal `performance` CPU governor.
- **Mouse and touchpad control:** The `libinput` library is now used by default instead of `synaptics` for mouse and touchpad control, with important improvements for the logic of disabling laptop trackpads while typing.
- **Snapper disk usage limits:** The Snapper configuration has been modified from its defaults to not use more than 40% of the available disk space for its automatic Btrfs snapshots[^1] – 20% maximum for system rollback snapshots and 20% maximum for snapshots of user data in `/home/` .
- **Font rendering and theming:** For each edition, many intricate details were analyzed and refined related to theming of GTK3, GTK4, and Qt apps, taking into account the unique characteristics and limitations of each toolkit running on a specific desktop environment, as well as the unique challenges of theming Flatpak apps created in those same three toolkits. Beyond simple aesthetic preferences, special priority was given to font legibility, contrast, and rendering, as well as overriding modern trends in all three toolkits that are detrimental to usability, such as thin and disappearing scrollbars without up/down arrow controls. The end result, while not completely uniform, is visually coherent and highly legible and usable even with a mix of Debian packaged apps and Flatpaks using different toolkits.
- **Other minor improvements and fixes:** [Issue tracker](https://github.com/SpiralLinux/SpiralLinux-project/issues?q=is%3Aissue+label%3Aapplied+updated%3A%3E2023-01-01+)

## How to upgrade SpiralLinux
Once installed to the hard disk the system no longer has any connection with the SpiralLinux project, as it has been carefully designed to be completely dependent on the Debian project and not on a single developer. Therefore, users of existing SpiralLinux installations can upgrade to the latest Debian release as soon as it is available in the Debian repositories. Most aspects of the unique SpiralLinux configuration will be maintained across upgrades, but new SpiralLinux features will not be applied to existing installations as there are intentionally no package repositories created specifically for SpiralLinux. To upgrade a SpiralLinux 11 installation to Debian 12, or to move to Debian Testing or Debian Unstable, please follow these simple graphical instructions:
https://github.com/SpiralLinux/SpiralLinux-project/wiki#updating-and-upgrading-spirallinux
> ⚠️ ***Important note:*** The Debian 12 version of the "Software & Updates" GUI tool has a significant unhandled bug that prevents it from enabling the Debian Security updates repository.[^2] [^3] Users that have upgraded to Debian 12 with Synaptic and the "Software & Updates" GUI should use the file manager's "Edit as root" function that is provided by default in all editions of SpiralLinux to replace the contents of `/etc/apt/sources.list` with the following:
```
deb https://deb.debian.org/debian/ bookworm main contrib non-free non-free-firmware

deb https://security.debian.org/ bookworm-security main contrib non-free non-free-firmware

deb https://deb.debian.org/debian/ bookworm-updates main contrib non-free non-free-firmware

deb https://deb.debian.org/debian/ bookworm-backports main contrib non-free non-free-firmware  #Debian Bookworm Backports
```

## Known issues
Many workarounds and fixes for upstream issues have been implemented in SpiralLinux and also in the deployments of Debian's `live-build` tool used to generate the ISOs.[^4] [^5] The following issues can not be corrected within the scope of the SpiralLinux project:
- **Budgie window manager theme:** In most cases after the initial login to the Budgie desktop the window manager will use the wrong theme for the window controls. Subsequent logins will use the correct theme.
- **_Software & Updates_ GUI:** As mentioned previously, the Debian 12 version of the "Software & Updates" GUI tool has a significant unhandled bug that prevents it from enabling the Debian Security updates repository.[^2] [^3]
- **Displayed memory usage:** Most programs that display the system RAM usage will now show a much higher number in Debian 12 compared to the same system running Debian 11. However, this is ***not*** indicative of drastically higher RAM consumption, but rather is due to a change in the way that memory usage is calculated.[^6]  The `htop` program still measures RAM usage according to the previous method.
- **Other long-standing issues:** [Issue tracker](https://github.com/SpiralLinux/SpiralLinux-project/labels/cantfix)

[^1]: SpiralLinux includes an advanced Btrfs subvolume layout by default with supporting tools that take automatic snapshots of user data and the system state, allowing for booting into snapshots and performing instant system rollbacks. While it is extremely rare for a Debian Stable update to break the system in a way that would require a rollback, it is much more likely to happen to users that have upgraded their SpiralLinux system to Debian Testing or Unstable (easy method [here](https://github.com/SpiralLinux/SpiralLinux-project/wiki#switching-from-debian-stable-to-the-testing-or-unstable-branch)). Additionally, the system rollback mechanism provides a layer of protection again system administrator mistakes, as well as taking automatic snapshots of user data in `/home/` in case of accidental file edits and/or deletions. Please read the [documentation](https://github.com/SpiralLinux/SpiralLinux-project/wiki#working-with-btrfs-snapshots-and-rollbacks) for easy instructions to perform a system rollback in SpiralLinux.
[^2]: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1023242
[^3]: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1040743
[^4]: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1051547
[^5]: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1018873
[^6]: https://bugzilla.redhat.com/show_bug.cgi?id=2003033

**Comment by zcatav** [comment]
Hi,
I'm on testing already.
My only repo source is /etc/apt/sources.list.d/spirallinux.list and contains just this line.

deb http://deb.debian.org/debian/ testing main non-free non-free-firmware contrib #Debian Testing

 I think I need no action?

Thanks for your great work.

**Reply by geckolinux** [reply]
Hi there, yes, that's correct.

**Comment by zcatav** [comment]
Many thanks.

Zeki Çatav
zekicatav.com

2 Eki 2023 Pzt 17:14 tarihinde GeckoLinux ***@***.***> şunu
yazdı:

**Comment by Adihashish** [comment]
Hi. I would like to switch from kubuntu to debian, but after installing spiral don't see any firewall configuration utility.

First installed gufw, set it on. But it's off on next reset

Secondly, tried the one from kubuntu kde control centre. Here i have to install it with discover which is fine. While in kubuntu i set it on once and it's on, here, I can't see if it's on or not.

So if you can help me with firewall issue, I would be glad to switch from corporate to community linux.

p.s.Don't tell me to go into command line and write zillions of commands. I won't do it out of principle. I have life, I have sex I have friends. No time for nerdy things. I expect my desktop distro to have graphical tools to fine tune my desktop. Kde has everything, in kubuntu even has firewall module in settings. In debian i can install graphical firewall setup tool but I want to be sure that it's on

Thank you

P.p.s. tested spiral via ventoy and rufus and it works

**Reply by geckolinux** [reply]
Hi there,

> I have friends. No time for nerdy things

In that case you probably don't need a firewall. Unless you're running SpiralLinux on a personal computer with a public IP address (extremely uncommon) then you're already behind your router's firewall, and plus the the translation between public and private IP addresses further isolates your computer.

> First installed gufw, set it on. But it's off on next reset

Sorry, but I just tested it and it works for me.

> P.p.s. tested spiral via ventoy and rufus and it works

It may "work" for some, but not for others, and those utilities tend to mess with the default boot parameters, which causes problems with some preconfiguration items that need to run on live boot.

**Reply by Adihashish** [reply]
ok thanks. Will try gufw again.

regardless of my activities, it is wise to have firewall on all the time, don't you think?
I'm torrenting a lot, mostly new movies with dht on. So firewall must be on

**Reply by geckolinux** [reply]
> regardless of my activities, it is wise to have firewall on all the time, don't you think?

Nope. Please see:
https://distrowatch.com/weekly.php?issue=20230904#qa

**Reply by Adihashish** [reply]
lol distrowatch you must be joking.

Even if they are right by some miracle, the fact is that its my laptop wifi and i take it myself on travel. So its not a home computer and this "isp firewall" argument is irrelevant. Plus, I live in Croatia, here you can't trust your isp on anything.

edit: gufw works :)

**Comment by Adihashish** [comment]
by the way are you planning to update gecko linux? it's been more than a year since last images.

I don't like corporate linux in general but my woman is manager in big IT company which is partnered with suse, so I have sweet spot for them. Plus Yast is awesome

**Reply by geckolinux** [reply]
openSUSE is currently neglecting the Cinnamon packages in their repositories. At this point Debian Stable has a much newer version of Cinnamon than openSUSE Tumbleweed. I've even submitted newer Cinnamon packages to openSUSE, but the requests were ignored. So until that changes there will be no new GeckoLinux releases. Meanwhile if you want you can install GeckoLinux and apply the current openSUSE updates and it will work just fine.

**Reply by Adihashish** [reply]
well, I do have issues, kde behaves weird after updating almost 2000 packages. For example can't reset or turn off via desktop but have to go into command line. Or for example network manager applet dissapears from system tray and since I'm using laptop wifi its a dealbreaker.

ok i understand you, if you like cinnamon just as i like kde,fine. Will try spiral again, and we'll see. Kindly waiting for new geckos

**Comment by zcatav** [comment]
If you have both entry (as "non-free" and "non-free-firmware") in your repo
list, please remove "non-free" and update the system.

sudo apt clean
sudo apt update -m
sudo dpkg --configure -a
sudo apt install -f
sudo apt upgrade -y
sudo apt autoremove --purge -y

3 Eki 2023 Sal 17:28 tarihinde Adihashish ***@***.***> şunu
yazdı:

**Reply by geckolinux** [reply]
> If you have both entry (as "non-free" and "non-free-firmware") in your repo list, please remove "non-free" and update the system. sudo apt clean sudo apt update -m sudo dpkg --configure -a sudo apt install -f sudo apt upgrade -y sudo apt autoremove --purge -y 3 Eki 2023 Sal 17:28 tarihinde Adihashish ***@***.***> şunu yazdı:
> […](#)
> well, I do have issues, kde behaves weird after updating almost 2000 packages. 

@Adihashish was referring to an issue he is experiencing with GeckoLinux, or it may be an upstream openSUSE issue, it's hard to say. But the above procedure would not be my recommendation.

**Comment by Adihashish** [comment]
Me again.

I see that spiral has 6.4. kernel and 6.1 kernel as well. At that is great, it is always wise to keep LTS kernel as backup.

When I update system via discover, is it going to update latest and LTS kernels automatically or I need to go to synaptic to update kernels myself?

**Reply by geckolinux** [reply]
There are no automatic updates of any packages in SpiralLinux. This is the recommended update method:
https://github.com/SpiralLinux/SpiralLinux-project/wiki#updates-within-a-debian-stable-release-lifecycle

**Reply by Adihashish** [reply]
ok i get it, discover for flatpaks only.

So when i use synaptic, is it going to update kernels, both lts and latest? or do i have to manually select kernel update like in some other distros?

**Comment by zcatav** [comment]
Hi,
Firewall is important but a secure VPN is more essential for your mentioned
stuations.

Zeki Çatav
zekicatav.com

3 Eki 2023 Sal 20:38 tarihinde Adihashish ***@***.***> şunu
yazdı:

**Comment by banzaigtv** [comment]
I'll look into this. My main laptop has already moved on to Debian 12 KDE, but I still have SpiralLinux GNOME (old version) as a backup. I love KDE Plasma more than anything else at this time since the newest version is so much more advanced than GNOME. When I get my older laptop back up, then I will look into installing the new SpiralLinux with XFCE on it.

**Comment by SEOhio** [comment]
Hello,
I appreciate the work you have done on this project. I am using the XFCE version and have installed this on four machines, the installation goes flawlessly but for some reason after performing the update through Synaptic the 6.12 kernel does not configure properly, complains about a dependency issue ending in an error 1. I end up having to mark the 6.12 for complete removal and then it will configure properly. I don't think I am doing anything incorrectly , just wanted to bring it to your attention.
Can't wait until Trixie launches, looks interesting.
Thanks again for creating this version.

Steve

**Reply by geckolinux** [reply]
Hi there, thanks for reporting. It's probably this upstream Debian Backports / Linux / Broadcom driver incompatibility, which you can easily fix if you don't have any Broadcom WiFI hardware: https://github.com/SpiralLinux/SpiralLinux-project/issues/525#issuecomment-2611070178

**Comment by SEOhio** [comment]
Thank you for responding so quickly, I will check that info out
